src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java

```diff
@@ -142,14 +142,15 @@ public boolean isReady(@Nonnull JCloudsSlave slave) {
         },
         JNLP {
             @Override
-            public ComputerLauncher createLauncher(@Nonnull JCloudsSlave slave) {
+            public ComputerLauncher createLauncher(@Nonnull JCloudsSlave slave) throws IOException {
+                Jenkins.getActiveInstance().addNode(slave);
```

Comment:
Without this step, if `isReady` returns false, `JCloudsCleanupThread::destroyServersOutOfScope` would immediately reclaim the provisioned JNLP node.

I noticed the "Server JNLPNode run out of its scope.... Terminating" peppering my logs. 

```
if (scope != null && scope.isOutOfScope()) {
                        LOGGER.info("Server " + server.getName() + " run out of its scope " + scope + ". Terminating.");
                        AsyncResourceDisposer.get().dispose(new DestroyMachine(cloud.name, server.getId()));
```
```
@Override
        public boolean isOutOfScope() {
            // TODO it might not exist yet
            return Jenkins.getActiveInstance().getNode(specifier) == null;
        }
```

So borrowed the behavior from `JCLoudsCloud::doProvision` --I didn't realize this was an anti-pattern.